src/main/python/mlsearch/api_requester.py

```diff
@@ -10,18 +11,20 @@
 class APIRequest():
     """For handling the different Valid API requests."""
 
-    def __init__(self, source, query, init_idx, count):
+    def __init__(self, source, query, init_idx, count, next_page_token=None):
         """
         Initialization for the class.
 
         :param  source:     The API request destination.
         :param  query:      The query for searching.
         :param  init_idx:   The initial pagination index.
         :param  count:      The number of records to be fetched.
+        :param  next_page_token: The current page token for youtube API.
         """
 
         self.params = {'query':query, 'init_idx':init_idx, 
-                            'count':count, 'source': source}
+                            'count':count, 'source': source,
+                            'next_page_token': next_page_token}
```

Comment:
This one too. if we can specify that this token is for YouTube only, it may be great 🤔

Reply:
```suggestion
                            'y_next_page_token': next_page_token}
```